ingestion/src/metadata/sampler/processor.py

```diff
@@ -0,0 +1,179 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Data Sampler for the PII Workflow
+"""
+import traceback
+from copy import deepcopy
+from functools import lru_cache
+from typing import Optional, cast
+
+from sqlalchemy import MetaData
+from sqlalchemy.orm import DeclarativeMeta
+
+from metadata.generated.schema.entity.data.database import Database
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.entity.services.databaseService import DatabaseConnection
+from metadata.generated.schema.entity.services.ingestionPipelines.status import (
+    StackTraceError,
+)
+from metadata.generated.schema.entity.services.serviceType import ServiceType
+from metadata.generated.schema.metadataIngestion.databaseServiceAutoClassificationPipeline import (
+    DatabaseServiceAutoClassificationPipeline,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    OpenMetadataWorkflowConfig,
+)
+from metadata.ingestion.api.models import Either
+from metadata.ingestion.api.parser import parse_workflow_config_gracefully
+from metadata.ingestion.api.step import Step
+from metadata.ingestion.api.steps import Processor
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.profiler.orm.converter.base import ometa_to_sqa_orm
+from metadata.profiler.source.metadata import ProfilerSourceAndEntity
+from metadata.sampler.models import SampleConfig, SampleData, SamplerResponse
+from metadata.sampler.sampler_interface import SamplerInterface
+from metadata.utils.profiler_utils import get_context_entities
+from metadata.utils.service_spec.service_spec import import_sampler_class
+
+NON_SQA_DATABASE_CONNECTIONS = ("Datalake",)
+
+
+@lru_cache
+def _get_sqa_metadata(conn_type: str) -> Optional[MetaData]:
+    """Set sqlalchemy metadata"""
+    if conn_type not in NON_SQA_DATABASE_CONNECTIONS:
+        return MetaData()
+    return None
```

Comment:
Would this cache the same metadata object? If so we would need to need to remove the caching. We need to make sure we get a new Metadata() instance each time otherwise we get some conflicts in orm object names if I well remember.

Basically the metadata instance will hold the orm object. if you have the same table name but from different schema / database then you run into issues

Reply:
I was trying to replicate this behavior https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/profiler/source/database/base/profiler_source.py#L78 of the `ProfilerSource` "freezing" a single MetaData object when it was instantiated. If I got that right, the ProfilerSource is created once at the DB Service level, and then each table would reuse the same MetaData object?

Reply:
otherwise, I am just as happy to remove the caching if there's no downside